document-store/src/integrationTest/java/org/hypertrace/core/documentstore/ArrayFiltersQueryTest.java

```diff
@@ -0,0 +1,231 @@
+  }
+
+  private static void initializeAndConnectToMongo() {
+    mongo =
+        new GenericContainer<>(DockerImageName.parse("mongo:4.4.0"))
+            .withExposedPorts(27017)
+            .waitingFor(Wait.forListeningPort());
+    mongo.start();
+
+    Datastore mongoDatastore =
+        DatastoreProvider.getDatastore(
+            DatastoreConfig.builder()
+                .type(MONGO)
+                .connectionConfig(
+                    ConnectionConfig.builder()
+                        .type(MONGO)
+                        .addEndpoint(
+                            Endpoint.builder()
+                                .host("localhost")
+                                .port(mongo.getMappedPort(27017))
+                                .build())
+                        .build())
+                .build());
+    datastoreMap.put(MONGO_STORE, mongoDatastore);
+  }
+
+  private static void createCollectionData() throws IOException {
+    final Map<Key, Document> documents =
+        Utils.buildDocumentsFromResource("query/array_operators/galaxy.json");
+    datastoreMap.forEach(
+        (k, v) -> {
+          v.deleteCollection(ArrayFiltersQueryTest.COLLECTION_NAME);
+          v.createCollection(ArrayFiltersQueryTest.COLLECTION_NAME, null);
+          Collection collection = v.getCollection(ArrayFiltersQueryTest.COLLECTION_NAME);
+          collection.bulkUpsert(documents);
+        });
+  }
+
+  @AfterAll
+  public static void shutdown() {
+    mongo.stop();
+    postgres.stop();
+  }
+
+  private static class AllProvider implements ArgumentsProvider {
+    @Override
+    public Stream<Arguments> provideArguments(final ExtensionContext context) {
+      return Stream.of(Arguments.of(MONGO_STORE), Arguments.of(POSTGRES_STORE));
+    }
+  }
+
+  private static class MongoProvider implements ArgumentsProvider {
+    @Override
+    public Stream<Arguments> provideArguments(final ExtensionContext context) {
+      return Stream.of(Arguments.of(MONGO_STORE));
+    }
+  }
+
+  private static class PostgresProvider implements ArgumentsProvider {
+    @Override
+    public Stream<Arguments> provideArguments(final ExtensionContext context) {
+      return Stream.of(Arguments.of(POSTGRES_STORE));
+    }
+  }
+
+  @ParameterizedTest
+  @ArgumentsSource(MongoProvider.class)
+  void getAllSolarSystemsWithAtLeastOnePlanetHavingBothWaterAndOxygen(final String dataStoreName)
+      throws IOException, JSONException {
+    final Collection collection = getCollection(dataStoreName);
+
+    final Query query =
+        Query.builder()
+            .setFilter(
+                DocumentArrayFilterExpression.builder()
+                    .operator(ANY)
+                    // # Can pass in some alias to this?
+                    .arraySource(IdentifierExpression.of("additional_info.planets"))
+                    .filter(
+                        and(
+                            ArrayRelationalFilterExpression.builder()
```

Comment:
Seems like `ALL` or `Contains All` more meaningful operator here. But, I guess, we will add it later.

Reply:
Yes. Currently, wanted to have a functionality, and later we can introduce operators to shortcut the functionality.